clients/src/main/java/org/apache/kafka/clients/consumer/Consumer.java

```diff
@@ -72,6 +72,10 @@ public interface Consumer<K, V> extends Closeable {
     */
     void subscribe(Pattern pattern);
 
+    void subscribe(SubscriptionPattern pattern, ConsumerRebalanceListener callback);
+
+    void subscribe(SubscriptionPattern pattern);
```

Comment:
This method needs a link to the javadocs in `KafkaConsumer`:
```suggestion
    /**
     * @see KafkaConsumer#subscribe(SubscriptionPattern)
     */
    void subscribe(SubscriptionPattern pattern);
```